core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -51,6 +51,9 @@
     !(defined(KOKKOS_ARCH_KEPLER) || defined(KOKKOS_ARCH_MAXWELL50) ||  \
       defined(KOKKOS_ARCH_MAXWELL52))
 #include <cuda_fp16.h>
+#include <iosfwd>  // istream & ostream for extraction and insertion ops
```

Comment:
This isn't sufficient because you need the complete types for `basic_*stream` to define `operator<<` and `operator>>`.  That being said, you are probably getting it when you #include <string>.